xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1066,6 +1066,12 @@ int CGUIDialogVideoInfo::ManageVideoItem(const CFileItemPtr &item)
         result = RemoveItemsFromTag(item);
         break;
 
+      case CONTEXT_BUTTON_MARK_WATCHED:
+      case CONTEXT_BUTTON_MARK_UNWATCHED:
+        database.MarkAsWatched(*item, (button == (CONTEXT_BUTTON)CONTEXT_BUTTON_MARK_WATCHED));
```

Comment:
It has to go through `CVideoLibraryQueue` because otherwise you might interfer with another video library job that is writing to the database.